File: Python/백준/Gold/1394. 암호/암호.py
group = input()
password = input()
length = len(group)
data = {group[i]: i+1 for i in range(len(group))}

ans = 0
for i in range(len(password)):
    ans *= length
    ans += data[password[i]]
    ans %= 900528
print(ans)

File: Swift/백준/Gold/1394. 암호/암호.swift
import Foundation


let group = readLine()!
let password = readLine()!
let length = group.count

var data : [Character: Int] = [:]
for (i,ch) in group.enumerated() {
    data[ch] = i+1
}

var ans = 0

for ch in password {
    ans *= length

    if let value = data[ch] {
        ans += value
    }
    ans %= 900528
}

print(ans)